File: day_trader/sources/strategy/strategies/day_trader/strategy/services/websocket/common/DTO/order_routing/route_position_req.py
from sources.strategy.strategies.day_trader.business_entities.day_trading_position import *
from sources.strategy.strategies.day_trader.strategy.services.websocket.common.DTO.websocket_message import *
import json

class RoutePositionReq(WebSocketMessage):

    #region Static Attributes

    @staticmethod
    def _BUY():
        return "BUY"

    @staticmethod
    def _SELL ():
        return "SELL"

    #endregion

    def __init__(self, Msg,Side,ReqId,Qty, Type=None,Account=None,Symbol=None,PosId=None,Price=None,UUID=None):
        super(RoutePositionReq, self).__init__(Msg)
        self.Symbol=Symbol
        self.PosId = PosId
        self.Type=Type
        self.Side=Side
        self.Price = Price
        self.UUID = UUID
        self.ReqId = ReqId
        self.Qty = Qty
        self.Account = Account


    def toJSON(self):
        return json.dumps(self, default=lambda o: o.__dict__,sort_keys=True)

File: test_clients/C#/day_trader_testclient/DayTraderTestClient/DayTraderTestClient.Common/DTO/Order Routing/RoutePositionReq.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DayTraderTestClient.Common.DTO.Order_Routing
{
    public class RoutePositionReq : WebSocketMessage
    {
        #region Public Attributes

        public string UUID { get; set; }

        public string ReqId { get; set; }

        public string Symbol { get; set; }

        public int? PosId { get; set; }

        public string Side { get; set; }

        public int Qty { get; set; }

        public string Account { get; set; }

        public decimal? Price { get; set; }

        public decimal? StopLoss { get; set; }

        public decimal? TakeProfit { get; set; }

        public bool? CloseEndOfDay { get; set; }

        #endregion
    }
}
